game/game.py

```diff
@@ -493,6 +496,40 @@ def position_culled(self, pos: Point) -> bool:
                 return False
         return True
 
+    def iads_considerate_culling(self, tgo: TheaterGroundObject[Any]) -> bool:
+        if not self.settings.perf_do_not_cull_threat_IADS:
+            return self.position_culled(tgo.position)
+        else:
+            if self.settings.perf_culling:
+                if isinstance(tgo, EwrGroundObject):
+                    max_detection_range = tgo.max_detection_range().meters
+                    for z in self.__culling_zones:
+                        seperation = z.distance_to_point(tgo.position)
+                        # EWR detection range is very far.  If left at 100% range no EWR would ever be culled.
+                        if seperation < 0.25 * max_detection_range:
+                            return False
+                if isinstance(tgo, SamGroundObject):
+                    max_threat_range = tgo.max_threat_range().meters
+                    for z in self.__culling_zones:
+                        seperation = z.distance_to_point(tgo.position)
+                        # Need range so that SAM can be respected, but dont need huge respect if small SAM.
+                        respect_bubble = 1.15 * max_threat_range
+                        max_respect_range = (
+                            Distance.from_nautical_miles(16).meters + max_threat_range
+                        )
+                        min_respect_range = (
+                            Distance.from_nautical_miles(4).meters + max_threat_range
+                        )
+
+                        if respect_bubble > max_respect_range:
+                            respect_bubble = max_respect_range
+                        elif respect_bubble < min_respect_range:
+                            respect_bubble = min_respect_range
```

Comment:
I'm not clear on what this is trying to do. Plugging in some arbitrary numbers:

```
max_threat_range = 10000nm
respect_bubble = 11500nm
max_respect_range = 10016nm
min_respect_range = 10004nm
respect_bubble = max_respect_range
```

```
max_threat_range = 1nm
respect_bubble = 1.15nm
max_respect_range = 17nm
min_respect_range = 5nm
respect_bubble = mix_respect_range
```

```
max_threat_range = 100nm
respect_bubble = 115nm
max_respect_range = 116nm
min_respect_range = 104nm
# respect_bubble unchanged
```

What's the goal here? I understand wanting to buffer the range a bit to get SAMs that are _close_ to objectives, but why the max, and why does it scale with SAM size? I can also understand adding a minimum for things like SA-19s since those have such a small range but are usually near enough to catch something attacking the objective, but the buffer does that already as long as it's constant. Should this code just be:

```
bubble = max_threat_range + nautical_miles(4)
```
?

Reply:
So these numbers came from me doing a bit of trial and erroring for "feel", specifically, "oh crap I don't want to fly that direction any more," but this was my reasoning when I had written this:

Shorad threat zones' occupy a tiny area, and become near inconsequential when far from the target.
Long range SAM threat zones' occupy a large area, and it's quite quick to fly the minimum distance (4nm) when in the target area.

So I viewed a shorad who's 8 miles away as ignorable, but an SA-10 whose threat range starts 8 miles away as not ignorable.

An alternative to this approach would be to have a flat range and bump the range up from `4nm` to `16nm` or whatever feels right, but this ramping approach was me trying to still consider performance with culling.

Reply:
> So I viewed a shorad who's 8 miles away as ignorable, but an SA-10 whose threat range starts 8 miles away as not ignorable.

I think that's where we disagree. 8 miles from getting shot is 8 miles regardless if who's doing the shooting. I'd actually be _more_ worried about DHORAD if anything, since I have a very long time to get back out of range before the SA-10 hits me, whereas I may not even know that the SHORAD is shooting.

A constant size buffer makes more sense to me, but like I said I'm not the user here, so do take that as advice and not as a demand :)